doc/Analytics.xml

```diff
@@ -3397,127 +3397,71 @@ Added Recognition rule engines</revremark>
+        (barriers) with an optional direction. The configuration parameters also include the time
+        interval to report the events and time interval to reset its counter.<variablelist>
+          <varlistentry>
+            <term>Parameters</term>
+            <listitem>
+              <para role="param">Segments [tt:Polyline]</para>
+              <para role="text">One or more polylines used for detection. An object must have passed
+                through all of them to register as a count. ??? or ??? Passing through any of the
+                lines registers as a count. ???</para>
+              <para>ReportTimeInterval [xs:duration]</para>
+              <para>Optional time interval to report count information, starting from midnight. For
+                example, if the interval is set to PT2H, reports will be published at 2:00 am, 4:00
+                am, etc. Default is same as ResetTimeInterval.</para>
+              <para>ResetTimeInterval [xs:duration]</para>
+              <para>Periodic count reset time starting from midnight. For example, if the interval
+                is set to PT12H, the count will be reset at 12:00 noon and 12:00 midnight.</para>
+              <para>Direction [tt:Direction]</para>
+              <para>Optional count direction. See definition in <xref
+                  xmlns:xlink="http://www.w3.org/1999/xlink" linkend="_Ref529526650"/>. If Direction
+                is set to Any, Left will increment and Right will decrement the counter. Default is
+                Any.</para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>Topic</term>
+            <listitem>
+              <para role="param">tns1:RuleEngine/CountAggregation/Counter</para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>Source</term>
+            <listitem>
+              <para role="text">See <xref linkend="_msgSource"/></para>
+            </listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>Data</term>
+            <listitem>
+              <para role="param">Count [xs:int]</para>
+              <para role="text">Number of objects counted since last reset.</para>
+            </listitem>
+          </varlistentry>
+        </variablelist></para>
+      <para>CountAggregation defined by the following code using the rule description language: </para>
       <programlisting><![CDATA[<tt:RuleDescription Name="tt:CountAggregation">
   <tt:Parameters>
-    <tt:ElementItemDescription Name="LineSegments"
-      Type="tt:PolylineArrayConfiguration"/>
-    <tt:SimpleItemDescription  Name="ReportTimeInterval" Type="xs:duration"/>
-    <tt:SimpleItemDescription  Name="ResetTimeInterval"  Type="xs:duration"/>
-    <tt:SimpleItemDescription  Name="Direction"  Type="tt:Direction"/>
+    <tt:ElementItemDescription Name="Segments" Type="tt:Polyline"/>
```

Comment:
Is it tt:Polyline or tt:PolylineArray ? if its segments polyline array makes sense. Other option is to change this segments to just segment.

How can we notify the number of segments supported, using generic options ?

Instead of segments, from hanwha side we prefer just single line segment.

Reply:
Prefer to keep Segments as also used in LineDetector. My understanding is that one Polyline consists of a number of segments and hence the name Segments was chosen back in 2008.